src/SourceBuild/tarball/content/ArcadeOverrides/SourceBuildArcadeTools.targets

```diff
@@ -12,9 +12,9 @@
 
   <Target Name="CollectSourceBuildIntermediateNupkgDependencies"
           Condition="
-            ('$(ArcadeBuildFromSource)' == 'true' and
-              '$(ArcadeInnerBuildFromSource)' == 'true') or
-            '$(SetUpSourceBuildIntermediateNupkgCache)' == 'true'"
+            '$(DotNetBuildFromSourceFlavor)' != 'Product' and
+            (('$(ArcadeBuildFromSource)' == 'true' and '$(ArcadeInnerBuildFromSource)' == 'true') or
+              '$(SetUpSourceBuildIntermediateNupkgCache)' == 'true')"
```

Comment:
When do you expect this logic to run? in a repo's source-build leg only? I checked the binlogs here and it doesn't run during inner builds.

Reply:
It should only run in the repo's source-build leg.  This is what is happening today albeit in a slightly different form.  The changes I removed [here](https://github.com/dotnet/installer/pull/14224/files#diff-1156e2bbb8a51f57bf9f833f34a3044033a01b4b0dee011147d22beaf8665891L624) where conditioning out the inner `ReadSourceBuildIntermediateNupkgDependencies` task previously.  I am proposing a refactor of this logic to make it more readable.